sevntu-checks/src/main/java/com/github/sevntu/checkstyle/checks/design/StaticMethodCandidateCheck.java

```diff
@@ -527,10 +528,13 @@ else if (findFrameByName(frame, objCalledOn.getText()).isPresent()) {
      * @return true, if LITERAL_THIS is used or the usage is too complex to check.
      */
     private static boolean isIdentShouldBeChecked(DetailAST parentAst) {
-        final int parentAstType = parentAst.getType();
-        return parentAstType != TokenTypes.LITERAL_NEW
-                && parentAstType != TokenTypes.TYPE
-                && parentAstType != TokenTypes.METHOD_DEF;
+        final int[] parentTypesNotToBeChecked = {
```

Comment:
Move it to final field, to create it ones during whole execution of Check, not on each call of method

Reply:
Done.